### DISCUSSION TITLE
Provide derivatives in a Model

### DISCUSSION THREAD

**Discussion by lellid** [lmfit/lmfit-py#854] [Q&A] [answered]
Hi,

I'm looking for an example of a model (derived from Model), which provides not only the function itself, but also the derivatives with respect to the parameters. Is there any such example available? Is this functionality already implemented in the Model class?

In the docs, I found only one example, which does not use the Model class, [here](https://lmfit.github.io/lmfit-py/examples/example_fit_with_derivfunc.html). Please note that the model which I have in mind should use parameter constraints, too.

D.L.

**Comment by newville** [comment, marked as answer]
@lellid I believe we do not have an example of that.  In principle, it would be no different than using with Minimizer.   For all cases using lmfit, providing a jacobian function is complicated if bounds are placed on Parameters.  To handle that, the jacobian function would have to include the effect of the bounds transformation by chaining the derivatives.  I think this would not be impossible, and might even be "not hard", but I don't know of anyone who has actually done this, or if it could be automated easily.

**Comment by lellid** [comment]
Hi Matt, thanks for the answer. 
I already suspected that, because if I provided derivatives, the fit did not converge.
